src/handlers/pr_tracking.rs

```diff
@@ -164,3 +206,187 @@ WHERE r.user_id = $1;";
         .await
         .context("Update DB error")
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::config::Config;
+    use crate::github::{Issue, IssuesAction, IssuesEvent, Repository, User};
+    use crate::handlers::pr_tracking::{handle_input, parse_input, upsert_pr_into_workqueue};
+    use crate::tests::github::{default_test_user, issue, pull_request, user};
+    use crate::tests::{run_test, TestContext};
+    use tokio_postgres::GenericClient;
+
+    #[tokio::test]
+    async fn add_pr_to_workqueue_on_assign() {
+        run_test(|ctx| async move {
+            let user = user("Martin", 2);
+
+            run_handler(
+                &ctx,
+                IssuesAction::Assigned {
+                    assignee: user.clone(),
+                },
+                pull_request().number(10).call(),
+            )
+            .await;
+
+            check_assigned_prs(&ctx, &user, &[10]).await;
+
+            Ok(ctx)
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn remove_pr_from_workqueue_on_assign() {
+        run_test(|ctx| async move {
+            let user = user("Martin", 2);
+            set_assigned_prs(&ctx, &user, &[10]).await;
+
+            run_handler(
+                &ctx,
+                IssuesAction::Unassigned {
+                    assignee: user.clone(),
+                },
+                pull_request().number(10).call(),
+            )
+            .await;
+
+            check_assigned_prs(&ctx, &user, &[]).await;
+
+            Ok(ctx)
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn remove_pr_from_workqueue_on_pr_closed() {
+        run_test(|ctx| async move {
+            let user = user("Martin", 2);
+            set_assigned_prs(&ctx, &user, &[10]).await;
+
+            run_handler(
+                &ctx,
+                IssuesAction::Closed,
+                pull_request()
+                    .number(10)
+                    .assignees(vec![user.clone()])
+                    .call(),
+            )
+            .await;
+
+            check_assigned_prs(&ctx, &user, &[]).await;
+
+            Ok(ctx)
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn add_pr_to_workqueue_on_pr_reopen() {
+        run_test(|ctx| async move {
+            let user = user("Martin", 2);
+            set_assigned_prs(&ctx, &user, &[42]).await;
+
+            run_handler(
+                &ctx,
+                IssuesAction::Reopened,
+                pull_request()
+                    .number(10)
+                    .assignees(vec![user.clone()])
+                    .call(),
+            )
+            .await;
+
+            check_assigned_prs(&ctx, &user, &[10, 42]).await;
+
+            Ok(ctx)
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn ignore_issue_assignments() {
```

Comment:
This test confuses me...what's its purpose and how is it different than add_pr_to_workqueue_on_assign?

Reply:
I spent like 1 hour "fixing" PR tracking, because I thought that it also considers issues, not only PRs. Only after that I noticed `parse_input` :facepalm: So I thought that I would add an explicit test that PR tracking only considers PRs and not *issues*, because in code they are represented in an almost identical way.

Reply:
Added a comment.

Reply:
Oh I see, `issue()` instead of `pull_request()`, I didn't even notice.